api/base/filters.py

```diff
@@ -125,11 +127,13 @@ def query_params_to_odm_query(self, query_params):
 
         fields_dict = query_params_to_fields(query_params)
         if fields_dict:
-            query_parts = [
-                Q(self.convert_key(key=key), self.get_comparison_operator(key=key), self.convert_value(value=value, field=key))
-                for key, value in fields_dict.items() if self.is_filterable_field(key=key)
-            ]
-            # TODO Ensure that if you try to filter on an invalid field, it returns a useful error. Fix related test.
+            query_parts = []
+            for key, value in fields_dict.items():
+                if self.is_filterable_field(key=key):
+                    query = Q(self.convert_key(key=key), self.get_comparison_operator(key=key), self.convert_value(value=value, field=key))
+                    query_parts.append(query)
+                else:
+                    raise InvalidFilter()
             try:
                 query = functools.reduce(intersect, query_parts)
```

Comment:
Is the `intersect` method necessary? Why not directly use `operator.and_` in functools.reduce?

```
>>> import functools
>>> import operator
>>> from modularodm import Q
>>> a = Q('something', 'eq', 'nothing')
>>> b = Q('something2', 'eq', 'nothing')
>>> c = Q('something2', 'eq', 'everything')
>>> d = functools.reduce(operator.and_, [a,b,c])
>>> d
AND(RawQuery(something, eq, nothing), RawQuery(something2, eq, nothing), RawQuery(something2, eq, everything))
```